File: 345. Reverse Vowels of a String/solution.go
import "bytes"

func reverseVowels(s string) string {
	start := 0
	end := len(s) - 1
	sByte := []byte(s)
	set := []byte{'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}

	for start < end {
		f := bytes.IndexByte(set, sByte[start])
		e := bytes.IndexByte(set, sByte[end])
		if f >= 0 && e >= 0 {
			sByte[start], sByte[end] = sByte[end], sByte[start]
			start++
			end--
			continue
		}
		if f < 0 {
			start++
		}
		if e < 0 {
			end--
		}
	}

	return string(sByte)
}

File: 345. Reverse Vowels of a String/solution.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        return re.sub('(?i)[aeiou]', lambda m, v=re.findall('(?i)[aeiou]', s): v.pop(), s)

        # vowels = "aeiuoAEIOU"
        # start = 0
        # s = list(s)
        # end = len(s) - 1
        # while start < end:
        #     if s[start] not in vowels:
        #         start += 1
        #     if s[end] not in vowels:
        #         end -= 1
        #     if s[start] in vowels and s[end] in vowels:
        #         s[start], s[end] = s[end], s[start]
        #         start += 1
        #         end -= 1
        # return "".join(s)
